### PR TITLE
dotCMS/core#20245 fix automate dotcms-webcomponent lib publish

### PR DESCRIPTION
**Pull request by alfredo-dotcms**
### Proposed Changes
* automate dotcms-webcomponent lib publish

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
task https://github.com/dotcms/core/issues/20245
related to https://github.com/dotCMS/core-web/pull/1668/files

- Proposed convention 
image

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

### CODE REVIEW

File: .github/workflows/publish_webcomponents.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            - name: 'publish to npm'
+              run: |
+                  cd core-web
+                  npm set //registry.npmjs.org/:_authToken ${{ secrets.NPM_TOKEN }}
+                  cd dist/libs/dotcms-webcomponents
+                  npm publish
+            - name: Checkout Core & Bump webComponentsReleaseVersion, only on a release branch
+              env:
+                  # TODO: Change me to someone else's username
+                  GITHUB_USER: victoralfaro-dotcms
```
Comment (nollymar) [inline .github/workflows/publish_webcomponents.yml:46]: is it ok to have this username? shouldn't we use a company's generic username? How is it managed for others actions?

Comment (alfredo-dotcms) [reply .github/workflows/publish_webcomponents.yml:46]: this will be changed later

**Comment by alfredo-dotcms** [comment]
related branches and PRs
core: issue-automate-dotcms-ui-lib-generation - https://github.com/dotCMS/core/pull/20561/files
core-web: issue-automate-dotcms-ui-lib-generation - https://github.com/dotCMS/core-web/pull/1665/files
core-web: issue-20245-automate-dotwebcomponent-lib-generation - https://github.com/dotCMS/core-web/pull/1668/files
dev-scripts: issue-automate-dotcms-ui-lib-generation - https://github.com/dotCMS/dev-scripts/pull/3/files

**Review by nollymar — APPROVED** [review]

File: .github/workflows/publish_webcomponents.yml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+              id: filter
+              with:
+                  base: ${{ github.ref }}
+                  list-files: 'json'
+                  working-directory: 'core-web'
+                  filters: |
+                      dotcms-webcomponents:
+                        - 'libs/dotcms-webcomponents/**'
+
+            - name: Changes found
```
Comment (victoralfaro-dotcms) [inline .github/workflows/publish_webcomponents.yml:30]: This can be merged into a single step

File: .github/workflows/publish_webcomponents.yml [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+              uses: 'phips28/gh-action-bump-version@master'
+              with:
+                  commit-message: 'CI: bumps version to {{version}} [skip ci]'
+                  skip-tag: 'true'
+                  default: prerelease
+                  preid: 'rc'
+              env:
+                  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+                  PACKAGEJSON_DIR: 'core-web/libs/dotcms-webcomponents'
+            - name: 'Automated Version Bump - dotcms-webcomponents - is Master branch'
```
Comment (victoralfaro-dotcms) [inline .github/workflows/publish_webcomponents.yml:71]: Typo?

File: .github/workflows/publish_webcomponents.yml [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+                  path: 'core-web'
+            - name: Configuring Node.js
+              uses: actions/setup-node@v2-beta
+              with:
+                  node-version: '10.15.3'
+            - name: Installing dependencies
+              run: |
+                  cd core-web
+                  npm install
+            - name: 'Automated Version Bump - dotcms-webcomponents - is Release branch'
```
Comment (victoralfaro-dotcms) [inline .github/workflows/publish_webcomponents.yml:59]: Typo?

**Review by fmontes — APPROVED** [review]

**Pull request merged**